src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js

```diff
@@ -179,548 +172,601 @@ define(['js/logger',
-            };
+                //if it is an attribute it might be invalid according the current meta rules
+                if (this._isInvalidAttribute(selectedObjIDs, keyParts[0])) {
+                    dst[extKey].options = dst[extKey].options || {};
+                    dst[extKey].options.invalid = true;
+                }
 
-            _isInvalidAttribute = function (attrName) {
-                var i = selectionLength,
-                    validNames;
+                //if the attribute value is an enum, display the enum values
+                if (commonAttrMeta[key].enum && commonAttrMeta[key].enum.length > 0) {
+                    dst[extKey].valueItems = commonAttrMeta[key].enum.slice(0);
+                    dst[extKey].valueItems.sort();
+                }
 
-                while (i--) {
-                    cNode = _client.getNode(selectedObjIDs[i]);
-                    if (cNode) {
-                        validNames = cNode.getValidAttributeNames();
+                // Get the min max for floats and integers
+                if (dst[extKey].valueType === 'float' || dst[extKey].valueType === 'integer') {
+                    range = this._getAttributeRange(selectedObjIDs, keyParts[0]);
+                    dst[extKey].minValue = range.min;
+                    dst[extKey].maxValue = range.max;
+                }
+            } else if (isRegistry === true) {
+                //is it inherited??? if so, it can be reseted to the inherited value
+                if (this._isResettableRegistry(selectedObjIDs, keyParts[0])) {
+                    dst[extKey].options = dst[extKey].options || {};
+                    dst[extKey].options.resetable = true;
+                }
 
-                        if (validNames.indexOf(attrName) !== -1) {
-                            return false;
+                if (prefix === CONSTANTS.PROPERTY_GROUP_PREFERENCES + '.') {
+                    if (onlyRootSelected === false) {
+                        //decorator value should be rendered as an option list
+                        if (key === REGISTRY_KEYS.DECORATOR) {
+                            //dstList[extKey].valueType = "option";
+                            //FIXME: only the decorators for DiagramDesigner are listed so far
+                            dst[extKey].valueItems = decoratorNames;
+                        } else if (key === REGISTRY_KEYS.SVG_ICON || key === REGISTRY_KEYS.PORT_SVG_ICON) {
+                            dst[extKey].widget = PropertyGridWidgets.DIALOG_WIDGET;
+                            dst[extKey].dialog = DecoratorSVGExplorerDialog;
+                        }
+                    }
+                } else if (prefix === CONSTANTS.PROPERTY_GROUP_META + '.') {
+                    if (key === REGISTRY_KEYS.VALID_VISUALIZERS) {
+                        dst[extKey].value = dst[extKey].value === undefined ?
+                            '' : dst[extKey].value;
+                        dst[extKey].regex = '/[^\w\W]/';
```

Comment:
It's an old workaround for displaying the available plugins, visualizer etc. I'll make an issue for a multi-select space separated string value.